src/Language/Haskell/Expression/Parser.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# OPTIONS_GHC -fno-warn-unused-do-bind #-}
+module Language.Haskell.Expression.Parser where
+
+import           Control.Applicative        ((<$>), (<*), (*>), optional)
+import           Data.Char
+import           Language.Haskell.TH.Syntax
+import           Text.Parsec
+import           Text.Parsec.Language       (haskell, haskellStyle)
+import qualified Text.Parsec.Token          as P
+
+type SParser u v = Parsec String u v
```

Comment:
@jkarni: Do we ever use the `u` for something other than `()`? If not I would fix it to `()` here.